src/commands/install-packages.yml

```diff
@@ -74,7 +74,10 @@ steps:
               fi
         - restore_cache:
             keys:
-              - node-deps-<<parameters.cache-version>>-<<#parameters.include-branch-in-cache-key>>{{ .Branch }}-<</parameters.include-branch-in-cache-key>>{{ checksum "/tmp/node-project-lockfile" }}-{{ arch }}
+              - node-deps-{{ arch }}-<<parameters.cache-version>>-<<#parameters.include-branch-in-cache-key>>{{ .Branch }}-<</parameters.include-branch-in-cache-key>>-{{ checksum "package.json"}}-{{ checksum "/tmp/node-project-lockfile" }}
+              - node-deps-{{ arch }}-<<parameters.cache-version>>-<<#parameters.include-branch-in-cache-key>>{{ .Branch }}-<</parameters.include-branch-in-cache-key>>-{{ checksum "package.json"}}-
+              - node-deps-{{ arch }}-<<parameters.cache-version>>-<<#parameters.include-branch-in-cache-key>>{{ .Branch }}-<</parameters.include-branch-in-cache-key>>-
+              - node-deps-{{ arch }}-<<parameters.cache-version>>-
```

Comment:
I imagine it would be frustrating to specify that the branch be included in the cache key only to find that cache had been restored from a different branch. I could see restoring the cache from a different branch being harmful to performance in cases where a repository has many different branches that are wildly out of sync. This line really only benefits projects that should have specified that the branch not be included in the cache key but didn't at the expense of those who correctly accepted the default behavior of including the branch.

Reply:
You make a good point, thanks! This kind of goes back to the issue where we were saying it would be difficult to make this configurable to a certain level.

It seems like removing just line 80 should resolve this. That would leave line 79 to potentially respect both options.

Does that sound right to you (it selected multiple lines, so just making sure)?

Reply:
Yep, you nailed it; that sounds perfect to me.

(The convention I am familiar with is that comments refer to the last line that appears in the context pane on GitHub since that is the outcome of pressing the add comment button corresponding to that line. I could definitely believe different organizations have different conventions I am unfamiliar with though.)